src/app.js

```diff
@@ -78,6 +82,23 @@ class SAFEApp extends EventEmitter {
     return this._mutableData;
   }
 
+  /*
+  * Validates appInfo and properly handles error
+  */
+  validateAppInfo() {
+    const appInfo = this._appInfo;
+    const appInfoMustHaveProperties = ['id', 'name', 'vendor'];
+    let bool = false;
+    const hasCorrectProperties = appInfoMustHaveProperties.every((prop) => {
+      bool = Object.prototype.hasOwnProperty.call(appInfo, prop) && appInfo[prop];
```

Comment:
Shouldn't we also check that it's not an empty string after trimming white spaces from it?